### ISSUE TITLE
Glossary/Terms definitions

### ISSUE THREAD

**Issue by pombredanne** [clearlydefined/clearlydefined#30]
Here is some feedback on the new https://github.com/clearlydefined/clearlydefined/wiki/Glossary
(as an aside, if we use a plain .md file instead of the wiki, it becomes possible to have PRs and reviews: GH does not support PR on the Git wiki repo)

- Coordinates
About https://github.com/clearlydefined/clearlydefined/wiki/Glossary#coordinates : this terms is very much something borrowed from Java/Maven. Could we come up with something more neutral?

  -  type and provider: there is always a default provider for a given type, (e.g. the default public central registry for a package type)  so it is not needed by default. Also on the other hand if the provider (e.g. npmjs) is not the default for a type, then we need instead a fully qualifier URL to identify this. I suggest that we adopt the purl names that have been reviewed by many at https://github.com/package-url/purl-spec 

  - revision: version seems a much more common term than revision. This is also what's used in purl

- Component
About https://github.com/clearlydefined/clearlydefined/wiki/Glossary#component I thought we had discussed this and had agreed that we would use the term package for this rather than component?

**Comment by jeffmcaffer** [comment]
Thanks @pombredanne.  All fair points. Let me set some context here.

* Coordinates -- Are there any particular negatives associated with the term? FWIW, we were more or less unaware of the use of `coordinates` in Maven. (I probably knew about it in the back of my mind but its been years since interacting in that world). 

* type / provider -- yes, there is often a prevalent/common provider for a given type. It totally makes sense for user input affordances to make use of that and allow users to skip over specifying both. However under the covers, we would do well to be fully specified from the beginning. Looking at the purl spec it is unclear how something like an NPM hosted on GitHub would be represented. "type: should be the "form" of the thing (i.e., how you want to interpret the resource). GitHub is not a "form" it is a place.  However, if the structure of the purl is dictated by the schema/type then you'd have to use GitHub as the form of versioning on GitHub is not the same as what they use in NPM.

consider something NPM-ish (I just made that up, not sure what the exact purl form would be):
``` 
npm:%40foo/bar@1.0?provider=github
```
`@foo` the scope of the npm. but that does not help us locate the npm on github as the org/repo names may have no correspondence at all to the npm scope/name.  1.0 is the version of the npm. while there may be a corresponding tag in GitHub, it may also be v1.0 or non-existent or pointing to the wrong place/moved.

Look at it GitHub-ish:
```
github:myOrg/myrepo@342987fdea987?type=npm
```
OK, that gets me there and is more or less equivalent to ClearlyDefined coordinates but now `type` (i.e., form of the entity pointed to) is an afterthought and the form/name of that param is up to the purl `type` definer.  They may not even know that people put NPMs on GitHub (for example) and not even think to include an option to put a "type" param in.

This is more or less equivalent to the `content-type` problem in HTTP. you have a URL that gets you to some bytes but need to know how to interpret the bytes. If purl had declared and canonical way of specifying the `type` of the bytes indicated by the purl then it would be equivalent to CD coordinates. As it is, that `type` is crucial to the processing and structuring of ClearlyDefined so we can't leave it be ambiguous.

If that were the case then I'd strongly recommend not using `type` in purl to mean the scheme. Perhaps just use `scheme` since that part of the structure already has a name.

* As for `package` vs `component`, naming is hard. We started out with package but very consistently we got tripped up with people wanting to talk about source and `package` was just way too confusing. `component` was suitably ambiguous so as to be a fancy term for `thing` -- that is, just a placeholder.
